gamemodes/terrortown/gamemode/shared/sh_admincheckwatch.lua

```diff
@@ -0,0 +1,68 @@
+---
+-- Lists hooks added to GM:TTT2AdminCheck on startup, and locks the hook from further additions after initialization.
+-- @author Wryyyong
+-- @module TTT2AdminCheckWatch
+
+local hook = hook
+local table = table
+local print = print
+local debug = debug
+local concommand = concommand
+
+local TTT2AdminCheckLocked = TTT2AdminCheckLocked or nil
+TTT2AdminCheckWatch = TTT2AdminCheckWatch or {}
+
+-- Copy the existing hook.Add function to oldHookAdd if the latter does not already exist.
+local oldHookAdd = oldHookAdd or hook.Add
+
+---
+-- Checks the contents of event_name and whether or not initialization has been completed before passing the arguments onto the original function.
+-- @param string event_name The name of the event to hook onto
+-- @param string name The unique ID for the hook
+-- @param function func The function to execute when the hook is called
+-- @realm shared
+function hook.Add(event_name, name, func)
```

Comment:
Normally we use names without underscore.


```suggestion
function hook.Add(eventName, name, func)
```

Reply:
`event_name` was used simply to mirror the vanilla `hook.Add` params, nothing more